### ISSUE TITLE
"Module keras not found", but config is correct and files are where they need to be

### ISSUE THREAD

**Issue by jeroenclaes** [rstudio/keras#445]
After a successful `install_keras`, I get the following errors. 

```
Installation complete.

> tok<- text_tokenizer(100)
Using TensorFlow backend.
Error: TypeError: __init__() got an unexpected keyword argument 'file'

Detailed traceback: 
  File "/home/jeroen/.virtualenvs/r-tensorflow/lib/python2.7/site-packages/keras/__init__.py", line 3, in <module>
    from . import utils
  File "/home/jeroen/.virtualenvs/r-tensorflow/lib/python2.7/site-packages/keras/utils/__init__.py", line 6, in <module>
    from . import conv_utils
  File "/home/jeroen/.virtualenvs/r-tensorflow/lib/python2.7/site-packages/keras/utils/conv_utils.py", line 9, in <module>
    from .. import backend as K
  File "/home/jeroen/.virtualenvs/r-tensorflow/lib/python2.7/site-packages/keras/backend/__init__.py", line 87, in <module>
    from .tensorflow_backend import *
  File "/home/jeroen/.virtualenvs/r-tensorflow/lib/python2.7/site-packages/keras/backend/tensorflow_backend.py", line 5, in <module>
    import tensorflow as tf
  File "/home/jeroen/.virtualenvs/r-tensorflow/lib/python2.7/site-packages/tensorflow/__init__.py", line 24, in <module>
    from tensorflo
````

When I run the same code again, that error disappears and it is replaced with the following error:

```
Error: Python module keras was not found.

Detected Python configuration:

python:         /home/jeroen/.virtualenvs/r-tensorflow/bin/python
libpython:      /usr/lib/python2.7/config-x86_64-linux-gnu/libpython2.7.so
pythonhome:     /usr:/usr
virtualenv:     /home/jeroen/.virtualenvs/r-tensorflow/bin/activate_this.py
version:        2.7.15rc1 (default, Apr 15 2018, 21:51:34)  [GCC 7.3.0]
numpy:          /home/jeroen/.virtualenvs/r-tensorflow/local/lib/python2.7/site-packages/numpy
numpy_version:  1.14.5
keras:          /home/jeroen/.virtualenvs/r-tensorflow/local/lib/python2.7/site-packages/keras

python versions found: 
 /home/jeroen/.virtualenvs/r-tensorflow/bin/python
 /usr/bin/python
 /usr/bin/python3

```

So, it appears that R-keras cannot find python-keras. However, when I do a `ls` of `/home/jeroen/.virtualenvs/r-tensorflow/local/lib/python2.7/site-packages/keras` I can see a bunch of python files, and it looks like a correctly installed python package.  When I check the other directories listed in the python config, everything appears to be where it needs to be. 

I've tried removing python-keras and tensorflow from my system-wide python installation. 
I've tried to install everything as `sudo`. 
I've tried to set all R System variables related to python to "" .

None of these seem to fix the issue. 

So, any thoughts? 

Thanks in advance!

**Comment by th2018** [comment]
I have a similar problem. R studio cannot find python keras:

**Error: Python module keras was not found.

Detected Python configuration:

python:         /anaconda3/envs/r-tensorflow/bin/python
libpython:      /anaconda3/envs/r-tensorflow/lib/libpython3.7m.dylib
pythonhome:     /anaconda3/envs/r-tensorflow:/anaconda3/envs/r-tensorflow
version:        3.7.0 (default, Jun 28 2018, 07:39:16)  [Clang 4.0.1 (tags/RELEASE_401/final)]
numpy:          /anaconda3/envs/r-tensorflow/lib/python3.7/site-packages/numpy
numpy_version:  1.15.0
keras:          /anaconda3/envs/r-tensorflow/lib/python3.7/site-packages/keras

python versions found: 
 /anaconda3/envs/r-tensorflow/bin/python
 /usr/bin/python
 /usr/local/bin/python3
 /anaconda3/envs/my_env/bin/python**

Hope someone can help. Thank you so much!

**Comment by huyunwei** [comment]
Ran into same problem. 
It appears that when R Keras calls Conda, Python 3.7 will be installed. (as of today.) 

I created the "r-tensorflow" environment with Python 3.6, and installed tf and keras. R can find the keras, and works fine.  

It should be an easy fix to force conda to install Python 3.6

**Comment by skeydan** [comment]
If you use the github version of keras & tensorflow

```
devtools::install_github("rstudio/tensorflow")
devtools::install_github("rstudio/keras")
```
with the conda installation method, it will install Python 3.6
